sherpa/astro/ui/tests/test_astro_supports_pha2.py

```diff
@@ -0,0 +1,363 @@
+    # (if any)
+    #
+    for i in range(1, 13):
+        assert i in pha_ids
+
+    for i in range(1, 13):
+        d = ui.get_data(i)
+        validate_pha(d, bkg=True)
+
+        # There is no indication of what "part" this data set
+        # represents in the file name
+        #
+        assert d.name == infile
+
+        b = ui.get_bkg(i, bkg_id=1)
+        validate_pha(b, bkg=False)
+        assert b.name == infile
+
+        b = ui.get_bkg(i, bkg_id=2)
+        validate_pha(b, bkg=False)
+        assert b.name == infile
+
+    ui.clean()
+
+
+# TODO: how to test what messages are logged when load_pha is
+#       called with a pha2 file?
+
+
+@requires_data
+@requires_fits
+def test_load_pha2_compare_meg_order1(make_data_path):
+    """Do we read in the MEG +/-1 orders?"""
+
+    # The MEG -1 order is dataset 9
+    # The MEG +1 order is dataset 10
+    #
+    pha2file = make_data_path('3c120_pha2')
+    meg_p1file = make_data_path('3c120_meg_1.pha')
+    meg_m1file = make_data_path('3c120_meg_-1.pha')
+
+    ui.clean()
+    ui.load_pha('meg_p1', meg_p1file)
+    ui.load_pha('meg_m1', meg_m1file)
+
+    orig_ids = set(ui.list_data_ids())
+    assert 'meg_p1' in orig_ids
+    assert 'meg_m1' in orig_ids
+
+    ui.load_pha(pha2file)
+
+    for n, lbl in zip([9, 10], ["-1", "1"]):
+        h = '3c120_meg_{}'.format(lbl)
+        ui.load_arf(n, make_data_path(h + '.arf'))
+        ui.load_rmf(n, make_data_path(h + '.rmf'))
+
+    # check that loading the pha2 file doesn't overwrite existing
+    # data
+    new_ids = set(ui.list_data_ids())
+
+    for i in range(1, 13):
+        orig_ids.add(i)
+
+    assert orig_ids == new_ids
+
+    # Check that the same model gives the same statistic
+    # value; this should check that the data and response are
+    # read in, that grouping and filtering work, and that
+    # model evaluation is the same, without having to
+    # check these steps individually.
+    #
+    # The model is not meant to be physically meaningful,
+    # just one that reasonably represents the data and
+    # can be evaluated without requiring XSPEC.
+    #
+    pmdl = ui.create_model_component('powlaw1d', 'pmdl')
+    pmdl.gamma = 0.318
+    pmdl.ampl = 2.52e-3
+
+    ncts = 20
+    for i in [9, 10, "meg_m1", "meg_p1"]:
+        ui.set_analysis(i, 'wave')
+        ui.group_counts(i, ncts)
+        ui.notice_id(i, 2, 12)
+        ui.set_source(i, pmdl)
+
+    ui.set_stat('chi2datavar')
+    s9 = ui.calc_stat(9)
+    s10 = ui.calc_stat(10)
+
+    sm1 = ui.calc_stat('meg_m1')
+    sp1 = ui.calc_stat('meg_p1')
+
+    # Since these should be the same, we use an equality test
+    # rather than approximation. At least until it becomes
+    # a problem.
+    #
+    assert s9 == sm1
+    assert s10 == sp1
+
+    # The values were calculated using CIAO 4.9, Linux64, with
+    # Python 3.5.
+    #
+    assert s9 == pytest.approx(1005.4378559390879)
+    assert s10 == pytest.approx(1119.980439489647)
+
+    ui.clean()
```

Comment:
I am a little concerned by these `clean` statements, which are not executed unless the tests pass. I'd rather create a fixture for it, to make sure `clean` is called before and after a test, whether or not it passes. I have been thinking about an automatic fixture to run before and after every test, but I don't want to stir the pot too much, so maybe for now it is enough to just include the fixture when needed.

Reply:
I have a simple fixture that does this in https://github.com/sherpa/sherpa/pull/412/commits/35c228774669ec932b92da35f4cd73ebe8487fb2 - we might want two versions, or be able to parametrize whether it's sherpa.ui or sherpa.astro.ui being used.

However, for these particular tests, I wonder if we should just change to creating a session object within the test and use that. I'm using the session layer both as a test of the high-level functionality (i.e. that it all works together), and because it's quicker to write the test at this level then have  a bunch of tests that check whether a particular class (e.g. Fit or Stat or Model or ..) works here, since really we should rely on the tests for those components.